api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -252,6 +253,13 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
       runtime = new Runtime();
     }
 
+    GceWithPdConfig gceWithPdConfig = runtime.getGceWithPdConfig();
+    if (gceWithPdConfig != null) {
+      PersistentDiskRequest persistentDiskRequest = gceWithPdConfig.getPersistentDisk();
+      if (persistentDiskRequest != null && persistentDiskRequest.getName().isEmpty()) {
+        persistentDiskRequest.setName(userProvider.get().generatePDName());
```

Comment:
We could consider adding validation here to prevent them from creating a new PD if they still have an active PD in the project. This wouldn't be transactionally tight though, so it's only an incremental improvement.

It also might be too slow - just an idea for a future improvement.